### PR TITLE
FOGL-3640: addition of delivery queues

### PR DESCRIPTION
**Pull request by pintomax**
FOGL-3640: addition of delivery queues

### CODE REVIEW

**Comment by pintomax** [comment]
Test tests/system/python/e2e/test_e2e_notification_service_with_plugins.py --wait-time=10 --retries=10
passed

File: C/services/common/delivery_queue.cpp [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+ * then we know then the plugin has been removed at notification instance shutdown
+ *
+ * After this particular delivery data element there will be no more entries in the delivery queue.
+ *
+ * @param   elem	Element in the queue
+ */
+void DeliveryQueue::processDelivery(DeliveryQueueElement* elem)
+{
+
+	if (elem->getName().compare("deliveryB") == 0)
```
Comment (MarkRiddoch) [inline C/services/common/delivery_queue.cpp]: What's the magic are about?

**Review by MarkRiddoch — COMMENTED** [review]
Looked good but I wasn't;t sure what the magic around "delivery" was all about.

File: C/services/common/delivery_queue.cpp (hunk 1 above)
Comment (pintomax) [reply C/services/common/delivery_queue.cpp]: I left a debug code, removed

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**